src/test/java/net/hurstfrost/jenkins/avatar/user/AvatarPropertyTest.java

```diff
@@ -0,0 +1,191 @@
+import org.kohsuke.stapler.StaplerRequest2;
+
+@WithJenkins
+class AvatarPropertyTest {
+
+    public static final String TEST_USER_NAME = "avatar";
+
+    public static final String AVATAR_FILE_NAME = "avatar.png";
+    public static final Path TEST_AVATAR_PATH = Path.of("./src/test/resources/" + AVATAR_FILE_NAME);
+    public static final String AVATAR_CONTENT_TYPE = "image/png";
+
+    @Test
+    void getAvatarUrl(@SuppressWarnings("unused") JenkinsRule r) throws Exception {
+        User user = User.get(TEST_USER_NAME, true, Collections.emptyMap());
+
+        AvatarProperty avatarProperty = new AvatarProperty();
+        user.addProperty(avatarProperty);
+
+        // avatar property without avatar
+        assertNull(avatarProperty.getAvatarUrl());
+
+        // avatar property with avatar
+        byte[] buffer = Files.readAllBytes(TEST_AVATAR_PATH);
+        MockRequest request = new MockRequest(buffer, AVATAR_CONTENT_TYPE);
+        avatarProperty.reconfigure(request, null);
+        assertNotNull(avatarProperty.getAvatarUrl());
+    }
+
+    @Test
+    void isHasAvatar(@SuppressWarnings("unused") JenkinsRule r) throws Exception {
+        User user = User.get(TEST_USER_NAME, true, Collections.emptyMap());
+
+        AvatarProperty avatarProperty = new AvatarProperty();
+        user.addProperty(avatarProperty);
+
+        // avatar property without avatar
+        assertFalse(avatarProperty.isHasAvatar());
+
+        // avatar property with avatar
+        byte[] buffer = Files.readAllBytes(TEST_AVATAR_PATH);
+        MockRequest request = new MockRequest(buffer, AVATAR_CONTENT_TYPE);
+        avatarProperty.reconfigure(request, null);
+        assertTrue(avatarProperty.isHasAvatar());
+    }
+
+    @Test
+    void doImage(@SuppressWarnings("unused") JenkinsRule r) throws Exception {
+        User user = User.get(TEST_USER_NAME, true, Collections.emptyMap());
+
+        AvatarProperty avatarProperty = new AvatarProperty();
+        user.addProperty(avatarProperty);
+
+        // avatar property without avatar
+        MockResponse response = new MockResponse();
+        avatarProperty.doImage(null, response);
+        assertNull(response.getContentType());
+
+        // avatar property with avatar
+        byte[] buffer = Files.readAllBytes(TEST_AVATAR_PATH);
+        MockRequest request = new MockRequest(buffer, AVATAR_CONTENT_TYPE);
+        avatarProperty.reconfigure(request, null);
+        response = new MockResponse();
+        avatarProperty.doImage(null, response);
+        assertEquals(AVATAR_CONTENT_TYPE, response.getContentType());
+
+        // avatar property with avatar throwing exception
+        response = new MockResponse() {
+            @Override
+            public void serveFile(
+                    StaplerRequest2 req, InputStream data, long lastModified, long contentLength, String fileName)
+                    throws IOException {
+                throw new IOException("Oh no!");
+            }
```

Comment:
this isn't used is it?

Reply:
It is in https://github.com/strangelookingnerd/avatar-plugin/blob/59b653a97a9d94992594ffbd1f2d20542ea900ee/src/main/java/net/hurstfrost/jenkins/avatar/user/AvatarProperty.java#L70

Reply:
I meant the IOException

Reply:
Not sure if I understand. The IOException is thrown during the test and caught by https://github.com/strangelookingnerd/avatar-plugin/blob/59b653a97a9d94992594ffbd1f2d20542ea900ee/src/main/java/net/hurstfrost/jenkins/avatar/user/AvatarProperty.java#L73